examples/developer-tools/src/controller.ts

```diff
@@ -318,27 +338,117 @@ export class Controller {
     const input = this.viewModel.fileUploadInput as HTMLInputElement;
     const file = input.files[0];
     if (!file) return;
-    this.loadFromFile(file);
+    this.loadFromFile(file).then(({successes, fails}) => {
```

Comment:
Any reason to destructure if you're only going to restructure it later?  (Maybe you need to make a shallow copy?  But if so then a comment noting this would be wise.)